adapter_test.go

```diff
@@ -255,6 +255,59 @@ func initAdapterWithGormInstanceByPrefixAndName(t *testing.T, db *gorm.DB, prefi
 	return a
 }
 
+func TestSoftDelete(t *testing.T) {
+	type TestCasbinRule struct {
+		CasbinRule
+		DeletedAt gorm.DeletedAt
+		//CreatedAt  time.Time
+	}
+
+	db, _ := gorm.Open(mysql.Open("root:@tcp(127.0.0.1:3306)/casbin"), &gorm.Config{})
+
+	a, _ := NewAdapterByDBWithCustomTable(db, &TestCasbinRule{}, "casbin_create_custom")
+
+	e, err := casbin.NewEnforcer("examples/rbac_model.conf", "examples/rbac_policy.csv")
+	if err != nil {
+		panic(err)
+	}
+
+	err = a.SavePolicy(e.GetModel())
+	if err != nil {
+		panic(err)
+	}
+
+	// Clear the current policy.
+	e.ClearPolicy()
+	testGetPolicy(t, e, [][]string{})
+
+	// Load the policy from DB.
+	err = a.LoadPolicy(e.GetModel())
+	if err != nil {
+		panic(err)
+	}
+	testGetPolicy(t, e, [][]string{
+		{"alice", "data1", "read"},
+		{"bob", "data2", "write"},
+		{"data2_admin", "data2", "read"},
+		{"data2_admin", "data2", "write"},
+	})
+
+	a.AddPolicy("p", "p", []string{"Carol", "data1", "read"})
```

Comment:
It is recommended to remove AddPolicy, so as to avoid AddPolicy failure and affect the test